Java/search/LinearSearch.java

```diff
@@ -0,0 +1,34 @@
+import java.util.*;
+
+public class LinearSearch {
+    //It iterates through the array to find key .The average time complexity O(n).
+    static int linSearch(int[] a,int k,int length){
+        for(int i=0;i<length;i++){
+            if(a[i]==k)
+                return i+1;//returns position
+        }
+        return -1;//returns -1 if not found
+    }
+    public static void main(String[] args){
+        Scanner sc = new Scanner(System.in);
+		System.out.print("Enter the length of the array: ");
+		int length = sc.nextInt();
+		int arr[] = new int[length];
+		System.out.print("Enter array values: ");
+		for(int i=0; i<length; i++) {
+			arr[i] = sc.nextInt();
+		}
+		System.out.print("Enter the value whose index needs to be searched: ");
+        int value = sc.nextInt();
+        System.out.println("Index of "+ value +" is: "+linSearch(arr,value,length));
+    }
+}
+/*
+Input:
```

Comment:
Add time and space complexity.

Reply:
I have added the complexities of the algorithm.